cmd/offlinesync/offlinesync.go

```diff
@@ -27,47 +27,55 @@ func Run(v *viper.Viper) (int, error) {
 	}
 
 	err = SyncOfflineActivity(v, queueFilepath)
-	if err != nil {
-		var errauth api.ErrAuth
-		if errors.As(err, &errauth) {
-			return exitcode.ErrAuth, fmt.Errorf(
-				"offline sync failed: invalid api key... find yours at wakatime.com/api-key. %s",
-				errauth,
-			)
-		}
-
-		var errbadRequest api.ErrBadRequest
-		if errors.As(err, &errbadRequest) {
-			return exitcode.ErrGeneric, fmt.Errorf(
-				"offline sync failed: bad request: %s",
-				err,
-			)
-		}
-
-		var errapi api.Err
-		if errors.As(err, &errapi) {
-			return exitcode.ErrAPI, fmt.Errorf(
-				"offline sync failed: api error: %s",
-				err,
-			)
-		}
-
-		var errSyncDisabled ErrSyncDisabled
-		if errors.As(err, &errSyncDisabled) {
-			log.Debugln(err.Error())
-
-			return exitcode.Success, nil
-		}
+	if err == nil {
```

Comment:
In Go is recommended/good practice to first check if error happened and then do the rest of the code if no error. So Would be good to simply add a new type assertion for the ErrBackoff inside the block `if err != nil`.

Reply:
Ok, but I'll have to move the logic into a function because that way fails linter.